Dockerfile

```diff
@@ -1,7 +1,13 @@
 FROM ubuntu:16.04
 
-RUN apt-get update && apt-get install -y curl  libunwind8 libicu55
-RUN curl -LO https://github.com/PowerShell/PowerShell/releases/download/v6.0.0-alpha.9/powershell_6.0.0-alpha.9-1ubuntu1.16.04.1_amd64.deb
-RUN dpkg -i powershell_6.0.0-alpha.9-1ubuntu1.16.04.1_amd64.deb
+ENV POWERSHELL_VERSION="6.0.0-alpha.9"
 
-Entrypoint [ "powershell" ]
+RUN apt-get update && \
+    apt-get install -y curl libunwind8 libicu55 && \
+    curl -LO https://github.com/PowerShell/PowerShell/releases/download/v"$POWERSHELL_VERSION"/powershell_"$POWERSHELL_VERSION"-1ubuntu1.16.04.1_amd64.deb && \
+    dpkg -i powershell_"$POWERSHELL_VERSION"-1ubuntu1.16.04.1_amd64.deb && \
```

Comment:
This Dockerfile is much better than the current one but coming from a proper shell script would be better.
I played with \tools\download.sh in a Dockerfile this morning and it doesn't need sudo when used in a Dockerfile so I am not sure how it would be used as is in a Dockerfile. Also the clean up lines in the Dockerfile could be moved to the shell script:

    apt-get clean --quiet
    apt-get autoclean --quiet
    rm -rf /var/lib/apt/lists/* ./$package

The above is for Ubuntu. Since I am not familiar with Centos and Darwin packages I don't know what needs to be done for proper cleanup.